### PR TITLE
Match doc comments to method names

### LINKED ISSUE #4683: The doc comment in Attestation doesn't match the actual method name: `ResetPolicy` vs  `ResetAttestationPolicy`
https://github.com/Azure/azure-sdk-for-cpp/blob/72cdbaa0fed2ba847607a0a5f372d70f83c41f2d/sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client_models.hpp#L496
The method name is `RestAttestationPolicy`, and the operation in the swagger being called `ResetPolicy` isn't visible/relevant to the end customer.
https://github.com/Azure/azure-sdk-for-cpp/blob/72cdbaa0fed2ba847607a0a5f372d70f83c41f2d/sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_administration_client.hpp#L148

Other places to update:
https://github.com/Azure/azure-sdk-for-cpp/blob/72cdbaa0fed2ba847607a0a5f372d70f83c41f2d/sdk/attestation/azure-security-attestation/src/private/attestation_client_models_private.hpp#L246

@LarryOsterman - could you please assign this to the appropriate folks

### PR DESCRIPTION
**Pull request by RickWinter**
Fixes #4683

### CODE REVIEW

File: sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp [hunk 1]
```diff
@@ -252,18 +252,18 @@ namespace Azure { namespace Security { namespace Attestation { namespace Test {
...
      *
-     * Forwards to the `SetPolicyTest` and `ResetPolicyTest` with a non-present AttestationSigingKey
+     * Forwards to the `SetPolicyTest` and `ResetAttestationPolicyTest` with a non-present AttestationSigingKey
      * parameter.
      */
     void ModifyPolicyUnsecuredTest()
     {
       SetPolicyTest();
-      ResetPolicyTest();
+      ResetAttestationPolicyTest();
```
Comment (ahsonkhan) [inline sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp:261]: The test changes don't seem necessary (and might be a side effect of find/replace).

But if we are modifying `ResetPolicyTest`, we should also consider updating `SetPolicyTest` to match.

**Review by ahsonkhan — APPROVED** [review]
Other than updating `SetPolicyTest`, looks good.

File: sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp (hunk 1 above)
Comment (RickWinter) [reply sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp]: Agreed, I had reverted the test changes but missed this file.  Looking at it, there are a couple comments that make sense to keep.  I've undone the changes to tests while leaving the comments referring to the SDK methods

File: sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp [hunk 2]
```diff
@@ -410,7 +410,7 @@ namespace Azure { namespace Security { namespace Attestation { namespace Test {
       switch (testInfo.param.TestType)
       {
         case TestCaseType::GetPolicy:
-          testName += "GetPolicy";
+          testName += "GetAttestationPolicy";
```
Comment (ahsonkhan) [inline sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp:413]: Revert this
```suggestion
          testName += "GetPolicy";
```

**Review by LarryOsterman — APPROVED** [review]

**Pull request merged**